www/src/public/js/controllers/builder/main.js

```diff
@@ -333,48 +379,71 @@
             else {
                 baseStats.hazelnut = 5;
             }
-
+            
+            runeCharms.charm1 = "AAAAA";
+            runeCharms.charm2 = "AAAAA";
+            runeCharms.charm3 = "AAAAA";
+            runeCharms.charm4 = "AAAAA";
+            
             var items = [];
             var itemIndex = 0;
             while (listStr.length > 0) {
+                let isLocked = false;
+                if (listStr[0] === '.') {
+                    isLocked = true;
+                    listStr.substr(1);
+                }
+                
                 if (listStr[0] === '_') {
                     items.push({
                         id: 0,
                         slot: $scope.slotOrder[itemIndex],
-                        locked: false
+                        locked: isLocked
                     });
                     listStr = listStr.substring(1);
                 }
-                else {
-                    if (listStr[0] === '.') {
-                        items.push({
-                            id: encoder.toNumber(listStr.slice(1,4)),
-                            slot: $scope.slotOrder[itemIndex],
-                            locked: true
-                        });
-                        listStr = listStr.substring(4);
+                else if (listStr[0] === '-') {
+                    items.push({
+                        id: -5,
```

Comment:
can you add a variable to the `$scope` called something like `$scope.runeCharmId = -5` at the top of the file, similar to the `$scope.listVer` and use that instead of writing -5 everywhere?

This is called a 'magic number'. It's a hard-coded constant that doesn't have an obvious meaning. If you, instead, use a variable, now it's very obvious what it is, because the variable name explains it, and you won't see that -5 all over the file and wonder what it is.

Reply:
added $scope.runeCharmId instead of "-5" everywhere.